src/PhpPact/Consumer/InteractionBuilder.php

```diff
@@ -82,39 +76,30 @@ public function willRespondWith(ProviderResponse $response): bool
     {
         $this->interaction->setResponse($response);
 
-        return $this->mockServerHttpService->registerInteraction($this->interaction);
+        return $this->pact->registerInteraction($this->interaction);
     }
 
     /**
      * {@inheritdoc}
      */
     public function verify(): bool
     {
-        return $this->mockServerHttpService->verifyInteractions();
+        return $this->pact->verifyInteractions();
     }
 
     /**
-     * Writes the file to disk and deletes interactions from mock server.
+     * Create mock server before verifying.
      */
-    public function finalize(): bool
+    public function createMockServer(): void
     {
-        // Write the pact file to disk.
-        $this->mockServerHttpService->getPactJson();
-
-        // Delete the interactions.
-        $this->mockServerHttpService->deleteAllInteractions();
-
-        return true;
+        $this->pact->createMockServer();
     }
 
     /**
      * {@inheritdoc}
      */
     public function writePact(): bool
     {
-        // Write the pact file to disk.
-        $this->mockServerHttpService->getPactJson();
-
         return true;
     }
```

Comment:
This method do nothing. Do we want to remove it as well (another breaking change)?

Reply:
Yes, remove the function.

Reply:
Removed